src/index.rs

```diff
@@ -261,11 +272,20 @@ impl Index {
         &mut outpoint_to_txid,
         &mut coinbase_inputs,
         &mut ordinal_ranges_written,
+        &mut outputs_in_block,
       )?;
     }
 
     height_to_hash.insert(&height, &block.block_hash())?;
 
+    // TODO: how to set default differently?
+    if outputs_traversed.is_empty()? {
+      outputs_traversed.insert("outputs_traversed", &0)?;
+    }
+    // TODO: error handling
+    let sum = outputs_in_block + outputs_traversed.get("outputs_traversed")?.unwrap();
+    outputs_traversed.insert("outputs_traversed", &sum)?;
```

Comment:
Maybe something like this:

```suggestion
    statistics.insert(Statistics::OutputsTraversed.into(), statistics.get(Statistics::OutputsTraversed.into())?.unwrap_or(0) + outputs_in_block)?;    
```

There might be borrow errors, if so, then you can break it into two expressions like you had.

Reply:
Did it like this because `From<fn(u64) -> Statistics {Statistics::OutputsTraversed}>` (and therefore `into()`) is not implemented for `u64`
Can/should I implement it above somewhere?
```rust
    statistics.insert(
      &(Statistics::OutputsTraversed as u64),
      &(statistics
        .get(&(Statistics::OutputsTraversed as u64))?
        .unwrap_or(0)
        + outputs_in_block),
    )?;
```

Reply:
Gotcha. I think the right thing to do is to implement `From` like so:

```rust
impl From<Statistic> for u64 {
    fn from(statistic: Statistic) -> Self {
        statistic as u64
    }
}
```

(And I guess the name should be `Statistic`.)

The reason to avoid `as` is that it's error prone. Even though `Statistics` is declared as `repr(u64)`, `as u8` will work, and will silently truncate. So it's good to avoid it and use `.into()` instead.

Reply:
Ok, good to know